core/src/main/java/com/graphhopper/routing/util/BikeCommonFlagEncoder.java

```diff
@@ -385,6 +387,12 @@ int getSpeed( OSMWay way )
         int speed = PUSHING_SECTION_SPEED;
         String highwayTag = way.getTag("highway");
         Integer highwaySpeed = highwaySpeeds.get(highwayTag);
+        // Override highway speed for otherwise classified as pushing sections, in case that the tagging is similar to highway=cycleway
+        if (way.hasTag("highway", pushingSectionsHighways) && !highwayTag.equals("steps"))
+        {
+            if ((way.hasTag("foot", "no")) || (way.hasTag("segregated", "yes") || (way.hasTag("bicycle", "designated") || way.hasTag("bicycle", "official"))))
```

Comment:
I would remove foot=no here as this does not imply (like the others) that bike is designated